src/peer_connection.cpp

```diff
@@ -1841,15 +1842,16 @@ namespace libtorrent {
 		}
 #endif
 
-		m_became_uninterested = aux::time_now();
-
 #ifndef TORRENT_DISABLE_LOGGING
 		peer_log(peer_log_alert::incoming_message, "NOT_INTERESTED");
 #endif
 		if (m_peer_interested)
+		{
 			m_counters.inc_stats_counter(counters::num_peers_up_interested, -1);
+			m_became_uninterested = aux::time_now();
```

Comment:
this is the main change, to not restart the timestamp on duplicate not-interested messages